### ISSUE TITLE
SUREFIRE_PLUGIN_ERROR

### ISSUE THREAD

**Issue by Jean-Baptiste-Lasselle** [allure-examples/allure-junit5-example#2]
[//]: # (
. Note: for support questions, please use Stackoverflow or Gitter**. 
. This repository's issues are reserved for feature requests and bug reports.
.
. In case of any problems with Allure Jenkins plugin** please use the following repository 
. to create an issue: https://github.com/jenkinsci/allure-plugin/issues
.
. Make sure you have a clear name for your issue. The name should start with a capital 
. letter and no dot is required in the end of the sentence. An example of good issue names:
.
. - The report is broken in IE11
. - Add an ability to disable default plugins
. - Support emoji in test descriptions
)

#### I'm submitting a ... 
  - [x ] bug report
  - [ ] feature request
  - [ ] support request => Please do not submit support request here, see note at the top of this template.

#### What is the current behavior?

I have experienced an ERROR in maven build  `./mvnw clean verify` , which is the following : 
```bash
Error: Could not find or load main class org.apache.maven.surefire.booter.ForkedBooter
```
* Which one will find in the build-generated `target/surefire-reports/*.dumpstream`
* Running  `mvnw` with the `-X` command does not provide any more useful infos

#### What is the motivation / use case for changing the behavior?

It's a bug, and I provide here the solution, which is very simple, one argument has to be added to the `maven-surefire-plugin` , in the root `pom.xml` (namely `-Djdk.net.URLClassPath.disableClassPathURLCheck=true`) : 

```xml
    <build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-surefire-plugin</artifactId>
                <version>2.22.1</version>
                <configuration>
                    <argLine>
                        -javaagent:"${settings.localRepository}/org/aspectj/aspectjweaver/${aspectj.version}/aspectjweaver-${aspectj.version}.jar" -Djdk.net.URLClassPath.disableClassPathURLCheck=true
                    </argLine>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>org.aspectj</groupId>
                        <artifactId>aspectjweaver</artifactId>
                        <version>${aspectj.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
            <plugin>
                <groupId>io.qameta.allure</groupId>
                <artifactId>allure-maven</artifactId>
                <version>2.9</version>
                <configuration>
                    <reportVersion>${allure.version}</reportVersion>
                    <allureDownloadUrl>https://repo.maven.apache.org/maven2/io/qameta/allure/allure-commandline/${allure.version}/allure-commandline-${allure.version}.zip</allureDownloadUrl>

                </configuration>
            </plugin>
        </plugins>
``` 

#### Please tell us about your environment:

Here is my JDK version, and I run on Debian 9 Stretch Cinnamon :
 
```bash
jbl@pc-alienware-jbl:~/demov$ java -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-8u181-b13-2~deb9u1-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
jbl@pc-alienware-jbl:~/demov$ ./mvnw -version
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T14:41:47-04:00)
Maven home: /home/jbl/.m2/wrapper/dists/apache-maven-3.6.0-bin/2dakv70gp803gtm5ve1ufmvttn/apache-maven-3.6.0
Java version: 1.8.0_181, vendor: Oracle Corporation, runtime: /usr/lib/jvm/java-8-openjdk-amd64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "4.9.0-7-amd64", arch: "amd64", family: "unix"
jbl@pc-alienware-jbl:~/demov$ javac -version
javac 1.8.0_181
jbl@pc-alienware-jbl:~/demov$ uname -a
Linux pc-alienware-jbl 4.9.0-7-amd64 #1 SMP Debian 4.9.110-3+deb9u2 (2018-08-13) x86_64 GNU/Linux
jbl@pc-alienware-jbl:~/demov$ 
```
 

#### Other information 

[//]: # (
. e.g. detailed explanation, stacktraces, related issues, suggestions 
. how to fix, links for us to have more context, eg. Stackoverflow, Gitter etc
)

**Comment by baev** [comment]
@Jean-Baptiste-Lasselle https://stackoverflow.com/questions/53010200/maven-surefire-could-not-find-forkedbooter-class

Looks like a bug in OpenJDK & Surefire Maven plugin

**Comment by baev** [comment]
@Jean-Baptiste-Lasselle is there something we should done for this issue, or it can be closed?

**Comment by Jean-Baptiste-Lasselle** [comment]
@baev I think it this issue maybe closed, If I leave a proper explanation here.

I here write my advise, and I think, best choice in this case : 

* We have here a problem of dependency management between test tech stack, and operating system dependencies under.
* Therefore this dependency management problem has to be addressed with Infrastructure As Code
* Today, we have containers and pipelines, so the first move of this issue's management, in each company, should be switching to OS distro like alpine, in containers, to get rid of the issue.

So basically, switch to a JDK ready container, and run everything inside there, is my advice's first move.

Thanks a lot  @baev
